internal/endpoints/awsrulesfn/partitions.go

```diff
@@ -2,208 +2,210 @@
 
 package awsrulesfn
 
-import (
-	"github.com/aws/smithy-go/ptr"
-)
+// GetPartition returns an AWS [Partition] for the region provided. If the
+// partition cannot be determined nil will be returned.
+func GetPartition(region string) *PartitionConfig {
+	return getPartition(partitions, region)
+}
 
 var partitions = []Partition{
 	{
 		ID:          "aws",
 		RegionRegex: "^(us|eu|ap|sa|ca|me|af)\\-\\w+\\-\\d+$",
 		DefaultConfig: PartitionConfig{
-			Name:               ptr.String("aws"),
-			DNSSuffix:          ptr.String("amazonaws.com"),
-			DualStackDNSSuffix: ptr.String("api.aws"),
-			SupportsFIPS:       ptr.Bool(true),
-			SupportsDualStack:  ptr.Bool(true),
+			Name:               "aws",
+			DnsSuffix:          "amazonaws.com",
+			DualStackDnsSuffix: "api.aws",
+			SupportsFIPS:       true,
+			SupportsDualStack:  true,
 		},
-		Regions: map[string]PartitionConfig{
+		Regions: map[string]RegionOverrides{
 			"af-south-1": {
 				Name:               nil,
-				DNSSuffix:          nil,
-				DualStackDNSSuffix: nil,
+				DnsSuffix:          nil,
```

Comment:
**nit**: `nil` is the default value, we could generate only the overrides and not have to explicitly set them. This would shorten the region maps and make the generated partitions slightly more readable as the overrides would be more clear.